src/projectile.py

```diff
@@ -98,11 +97,13 @@ class Beam(Projectile):
 
     beam_range: float
 
-    def setup(self):
-        self.beam_range = self.sprite.width
+    def setup(self, beam_range:float, explodes: bool = False):
+        self.beam_range = beam_range
+        self.explodes = explodes
 
     def update(self, delta_time: float):
-        pass
+        set_sprite_location(self.sprite, self.muzzle_location)
+        move_sprite_polar(self.sprite, self.sprite.width/2, self.muzzle_location[2])
```

Comment:
This code specifically calls `self.sprite.width/2` rather than range because in the future, if the beam is shortened due to collision, using range won't move it correctly